tests/test_str.cpp

```diff
@@ -26,112 +27,112 @@ TEST_CASE("Various string operation", "[string]") {
 
   {
     std::string haystack = "n";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::kmp::make_needle("n"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_knuth_morris_pratt_searcher("n"));
     REQUIRE(index == haystack.begin());
   }
 
   {
     std::string haystack = "nmnn";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::kmp::make_needle("nn"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_knuth_morris_pratt_searcher("nn"));
     REQUIRE(std::distance(haystack.begin(), index) == 2);
   }
 
   {
     std::string haystack = "nmnn";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::kmp::make_needle("mm"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_knuth_morris_pratt_searcher("mm"));
     REQUIRE(index == haystack.end());
   }
 
 
   {
     std::string haystack = "ABC ABCDAB ABCDABCDABDE";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::kmp::make_needle("ABCDABD"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_knuth_morris_pratt_searcher("ABCDABD"));
     REQUIRE(std::distance(haystack.begin(), index) == 15);
   }
 
 }
 
-TEST_CASE("Boyer-Moore  str search", "[str-search]") {
+TEST_CASE("Boyer-Moore str search", "[str-search]") {
 
   {
     std::string haystack = "n";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::bm::make_needle("n"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_boyer_moore_searcher("n"));
     REQUIRE(index == haystack.begin());
   }
 
   {
     std::string haystack = "nmnn";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::bm::make_needle("nn"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_boyer_moore_searcher("nn"));
     REQUIRE(std::distance(haystack.begin(), index) == 2);
   }
 
   {
     std::string haystack = "nmnn";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::bm::make_needle("mm"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_boyer_moore_searcher("mm"));
     REQUIRE(index == haystack.end());
   }
 
 
   {
     std::string haystack = "ABC ABCDAB ABCDABCDABDE";
-    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::bm::make_needle("ABCDABD"));
+    auto index = frozen::search(haystack.begin(), haystack.end(), frozen::make_boyer_moore_searcher("ABCDABD"));
     REQUIRE(std::distance(haystack.begin(), index) == 15);
   }
 
 }
 
-TEST_CASE("str str perf", "[str-search-perf]") {
-  std::string haystack = "ABC ABCDAB ABCDABCDABDE";
-  for(int i = 0; i < 10; ++i)
-    haystack = "AAAAAAAAAAAAAAAA" + haystack;
+template<typename Func>
+void testStringSearchPerf(char const *message, Func test) {
+  std::string haystack(100'000'000, 'A');
```

Comment:
That's... huge :-)

Reply:
Yes... don't be shy :-) ! Only `std:string::find` suffers from this. The goal is just to have significant amount of work to measure.